### PR TITLE
#831 Add feature to only show  stores within the name search modal and requisitions statuses

### PR DESCRIPTION
**Pull request by joshxg**
#831 

- Requisitions should only be made to a store, not just any supplier - we get errors otherwise!
- Also added requisitions status changes - should have been in a new branch ugh.

### CODE REVIEW

File: packages/system/src/Name/Components/NameSearchModal/NameSearchModal.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ interface NameSearchProps {
   open: boolean;
   onClose: () => void;
   onChange: (name: Name) => void;
-
+  onlyShowStores?: boolean;
```
Comment (joshxg) [inline packages/system/src/Name/Components/NameSearchModal/NameSearchModal.tsx:10]: there's no filter in the API to do this, so we need to do it in JS. Not such a big deal. Though, this boolean annoys me.

An `extraFilter` prop where we can pass a more generic filtering function was another thought, though getting a list of 'Internal suppliers' is relatively common.

I had a thought that there should just be more components. `CustomerSearchModal`, `SupplierSearchModal`, `InternalSupplierSearchModal` etc instead

File: packages/requisitions/src/SupplierRequisition/DetailView/Footer/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
+import {
```
Comment (joshxg) [inline packages/requisitions/src/SupplierRequisition/DetailView/Footer/Footer.tsx:1]: shifted this file and essentially made it actually work

File: packages/requisitions/src/SupplierRequisition/DetailView/Footer/StatusChangeButton.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
+import React, { useMemo, useState, useEffect } from 'react';
```
Comment (joshxg) [inline packages/requisitions/src/SupplierRequisition/DetailView/Footer/StatusChangeButton.tsx:1]: All very similar/exactly the same as the other status change buttons

File: packages/requisitions/src/SupplierRequisition/DetailView/DetailView.tsx [hunk 4]
```diff
@@ -10,7 +10,7 @@ import {
 } from '@openmsupply-client/common';
 import { useRequestRequisition, useIsRequestRequisitionDisabled } from '../api';
 import { Toolbar } from './Toolbar';
-import { Footer } from './Footer';
+import { Footer } from './Footer/Footer';
```
Comment (mark-prins) [inline packages/requisitions/src/SupplierRequisition/DetailView/DetailView.tsx:13]: line 29 - the disabled state on AppBarButtons is reversed

File: packages/system/src/Name/Components/NameSearchModal/NameSearchModal.tsx (hunk 1 above)
Comment (mark-prins) [reply packages/system/src/Name/Components/NameSearchModal/NameSearchModal.tsx:10]: agree - creating more components would help here.

**Review by mark-prins — APPROVED** [review]
👍  LGTM!

File: packages/requisitions/src/SupplierRequisition/DetailView/DetailView.tsx (hunk 4 above)
Comment (joshxg) [reply packages/requisitions/src/SupplierRequisition/DetailView/DetailView.tsx:13]: Ah thanks, oops! Lucky it doesn't do anything yet 😁

**Pull request merged**